script/fetch-issues.js

```diff
@@ -141,23 +192,132 @@ async function fetchNextPage(owner, name, isIssue, data, cursor, pageCount) {
         }
       };
       if (!isIssue) {
-        info.pullRequest = {
-          htmlUrl: node.url
+        let rawComments, comments;
+        // collect review comments
+        rawComments = node.reviews.nodes.map(node => node.comments.nodes);
+        rawComments = [].concat.apply([], rawComments);
+        // collect issue comments
+        rawComments = rawComments.concat(node.comments.nodes);
+
+        // fetch reactions only if there are reactions within that PR
+        // to reduce API hits.
+        // trick: a comment has received reaction(s) iff
+        // reactionGroup.createdAt attribute is not null
+        let commentsWithReactions = [];
+        const hasReactions = rawComments.reduce((has, rawComment) => {
+          return has || rawComment.reactionGroups.reduce((flag, node) => {
+            return (flag || node.createdAt) ? true : false;
+          }, false);
+        }, false);
+
+        if (hasReactions) {
+          console.log('pull request has reactions. repoOwner:', owner,
+            'repoName:', name, 'pullRequest number:', node.number);
+          const number = node.number;
+          const reviewCnt = Math.min(node.reviews.totalCount, 20);
+          const maxCommentsPerReview = Math.min(
+            node.reviews.nodes.reduce(
+              (max, review) => Math.max(max, review.comments.totalCount), 0),
+            100);
+          const commentCnt = Math.min(node.comments.totalCount, 100);
+          commentsWithReactions = await fetchReactionsOfPR(owner, name, number,
+            reviewCnt, maxCommentsPerReview, commentCnt);
+
+          // merge rawComments with commentsWithReactions
+          if (commentsWithReactions) {
+            rawComments.map((comment, index) => {
```

Comment:
Not sure what this is doing, do you mean `forEach` ?

`map` will create new array.

Reply:
Although this also works, I meant to use `forEach`. 😅